adal/log.py

```diff
@@ -79,7 +79,7 @@ def __init__(self, component_name, log_context):
         self.log_context = log_context
         self._logging = logging.getLogger(ADAL_LOGGER_NAME)
 
-    def _log_message(self, msg, **kwargs):
+    def _log_message(self, msg, kwargs):

```

Comment:
Thanks for reporting this issue. While your patch technically works, I would like to suggest a slightly different approach.

1. Keep the `**kwargs` as-is at `_log_message()`, because it kinda feels weird to use a conventional `kwargs` parameter without the `**`
2. Do NOT do kwargs.pop(...) inside `_log_message()`, because changing mutable input objects is usually an unexpected side effect. I would rather suggest to explicitly do pop(...) in each of the `warn()`, `info()` and `debug()`, even that would mean 3 duplicated lines.
3. Yea we missed the test case for this scenario in the first place. Sorry about that. Would you mind to provide one together with your fix?

Let me know if you would like to proceed with the adjustment. Thanks!